lib/util/DocumentExtractor.js

```diff
@@ -42,9 +42,31 @@ const extractors = ([
     methods: {
       extractFromRequest: request => {
         const documents = [];
-
-        for (let it = 0; it < request.input.body.ids.length; it++) {
-          documents.push({ _id: request.input.body.ids[it] });
+        if ( request.input.body
+          && request.input.body.ids
+          && Object.keys(request.input.body.ids).length
+        ) {
+          for (let it = 0; it < request.input.body.ids.length; it++) {
+            documents.push({ _id: request.input.body.ids[it] });
```

Comment:
`Array.map` is faster than `for (let ... )` and `Array.push` to create new array from array

```
$ node loop-create-array.js
   foreach x 60,450 ops/sec ±8.91% (87 runs sampled)
   for (const of) x 112,013 ops/sec ±1.97% (84 runs sampled)
   for (let i) x 113,350 ops/sec ±2.63% (82 runs sampled)
   map x 153,657 ops/sec ±4.45% (78 runs sampled)
 Fastest is map
```

https://gist.github.com/Aschen/6225c7bc9c67313f9eeb3571964984e5

Reply:
(Also apply for all the array you create)